Detectors/ITSMFT/ITS/macros/test/CheckTracks.C

```diff
@@ -46,15 +46,28 @@ void CheckTracks(Int_t nEvents = 10, TString mcEngine = "TGeant3") {
   o2::dataformats::MCTruthContainer<o2::MCCompLabel> *trkLabArr=nullptr;
   recTree->SetBranchAddress("ITSTrackMCTruth",&trkLabArr);
 
-  
+  Int_t tf=0, nrec=0;
+  Int_t lastEventID=-1;
   Int_t nev=mcTree->GetEntries();
   for (Int_t n=0;n<nev;n++) {
-    std::cout<<"Event "<<n<<'/'<<nev<<std::endl;
+    std::cout<<"\nMC event "<<n<<'/'<<nev<<std::endl;
     Int_t nGen=0, nGoo=0;
     mcTree->GetEvent(n);
     Int_t nmc=mcArr->size();
-    recTree->GetEvent(n);
-    Int_t nrec=recArr->size();
+
+    while (n > lastEventID) { // Cache a new reconstructed TF
```

Comment:
If for the last MC event no track was reconstructed, this will lead to infinite loop, I think one should add  ``while (n>lastEventID && tf<recTree->GetEntries())``